bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/Channel.java

```diff
@@ -200,12 +201,43 @@ public Configuration getConfiguration() {
     }
 
     /**
-     * Returns the channel properties
+     * Returns an immutable copy of the {@link Channel} properties.
      *
-     * @return channel properties (not null)
+     * @return an immutable copy of the {@link Channel} properties (not {@code null})
      */
     public Map<String, String> getProperties() {
-        return properties;
+        synchronized (this) {
+            return Collections.unmodifiableMap(new HashMap<>(properties));
+        }
+    }
+
+    /**
+     * Sets the property value for the property identified by the given name. If the value to be set is {@code null}
+     * then the property will be removed.
+     *
+     * @param name the name of the property to be set (must not be {@code null} or empty)
+     * @param value the value of the property (if {@code null} then the property with the given name is removed)
+     * @return the previous value associated with the name, or {@code null} if there was no mapping for the name
+     */
+    public @Nullable String setProperty(String name, @Nullable String value) {
```

Comment:
You might notice that the Channel class does not have any kind of setters as it is considered to be immutable. So returning an immutable map of properties is definitely an important fix, but we imho should not introduce any setters at the same time. Instead, we should probably rather enhance the ChannelBuilder. Wdyt?